emote/callbacks.py

```diff
@@ -374,23 +371,37 @@ def __init__(
         callbacks: List[Callback],
         path: str,
         checkpoint_interval: int,
-        optimizers: Optional[List[optim.Optimizer]] = None,
-        networks: Optional[List[nn.Module]] = None,
     ):
         super().__init__(cycle=checkpoint_interval)
-        self._cbs = callbacks
+
         self._path = path
         self._checkpoint_index = 0
-        self._opts: List[optim.Optimizer] = optimizers if optimizers else []
-        self._nets: List[nn.Module] = networks if networks else []
+
+        self._cbs = []
+        names = []
+        for cb in callbacks:
+            if hasattr(cb, "name"):
+                self._cbs.append(cb)
+                names.append(cb.name)
+            else:
+                warnings.warn(
+                    f"Checkpointer ignored {cb} because of not "
+                    f"having the 'name' field.",
+                    UserWarning,
+                )
```

Comment:
Wondering whether this should be an exception instead?

Also; what we've done historically is to have 

```
@property
def name(self) -> str:
    return self.__class__.__name__
```

on the callback. So everything has a name by default. That leads to weird issues with duplicated callbacks, so that's why some things have explicit names.